runtime/src/main/java/org/corfudb/runtime/view/replication/QuorumReplicationProtocol.java

```diff
@@ -71,14 +71,19 @@ public ILogData peek(Layout layout, long address) {
         log.trace("Peek[{}]: quorum {}/{}", address, numUnits, numUnits);
         try {
             ReadResponse readResponse = null;
-            CompletableFuture<ReadResponse>[] futures = new CompletableFuture[numUnits];
-            for (int i = 0; i < numUnits; i++) {
-                futures[i] = layout.getLogUnitClient(address, i).read(address);
+            try {
+                CompletableFuture<ReadResponse>[] futures = new CompletableFuture[numUnits];
+                for (int i = 0; i < numUnits; i++) {
+                    futures[i] = layout.getLogUnitClient(address, i).read(address);
+                }
+                QuorumFuturesFactory.CompositeFuture<ReadResponse> future =
+                        QuorumFuturesFactory.getQuorumFuture(new ReadResponseComparator(address),
+                                futures);
+                readResponse = CFUtils.getUninterruptibly(future, QuorumUnreachableException.class);
+            } catch (QuorumUnreachableException e) {
+                log.debug(e.getMessage(), e);
```

Comment:
Can you add some description here? "peek: Quorum unreachable: {}"

Reply:
Sure. Done.